arch/arm/core/cortex_m/irq_init.c

```diff
@@ -25,9 +25,14 @@
 
 void z_arm_interrupt_init(void)
 {
+#if defined(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)
```

Comment:
As a best practice, it's better to use `IS_ENABLED(X)` or the newer `IF_ENABLED(X, ...)` to `#if defined(X)` since it preserves indentation better.

```c
if (IS_ENABLED(CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER)) {
    ...
} else {
    ...
}
```

Some files are already littered with the classic `#if defined`, but given this file is relatively small lets start with the newer macros here.

Reply:
Since the prototype for```z_soc_irq_init``` is only defined if ```CONFIG_ARM_CUSTOM_INTERRUPT_CONTROLLER=y``` I believe this will result in a compiler warning at best or error at worst. I'll try it out but in this case I think ```ifdefs``` are appropriate.